internal/internal_task_pollers.go

```diff
@@ -164,6 +200,18 @@ func (wtp *workflowTaskPoller) PollTask() (interface{}, error) {
 
 // ProcessTask processes a task
 func (wtp *workflowTaskPoller) ProcessTask(task interface{}) error {
+	if lar, ok := task.(*localActivityResult); ok {
```

Comment:
Could we have two functions like ProcessTask and ProcessActivityResult which would delegate to some internal method for common parts. Having method that switches on the type of parameter is confusing.